### PR TITLE
PMM-10948 code review

### PR DESCRIPTION
**Pull request by YaroslavPodorvanov**
[PMM-10541](https://jira.percona.com/browse/PMM-10541) [PMM-10948](https://jira.percona.com/browse/PMM-10948)

https://github.com/Percona-Lab/go-grpc-prometheus/pull/3
https://github.com/percona/pmm/pull/1398

### CODE REVIEW

File: server_metrics.go [hunk 1]
```diff
@@ -62,15 +62,22 @@ func NewServerMetrics(counterOpts ...CounterOption) *ServerMetrics {
 // expensive on Prometheus servers. It takes options to configure histogram
 // options such as the defined buckets.
 func (m *ServerMetrics) EnableHandlingTimeHistogram(opts ...HistogramOption) {
-	for _, o := range opts {
```
Comment (ritbl) [inline server_metrics.go:65]: why you removed applying opts?

File: server_metrics.go [hunk 2]
```diff
@@ -62,15 +62,22 @@ func NewServerMetrics(counterOpts ...CounterOption) *ServerMetrics {
...
+	if m.serverHandledHistogramEnabled {
+		return
 	}
-	if !m.serverHandledHistogramEnabled {
-		m.serverHandledHistogram = prom.NewHistogramVec(
-			m.serverHandledHistogramOpts,
-			[]string{"grpc_type", "grpc_service", "grpc_method"},
-		)
+
+	serverHandledHistogramOpts := prom.HistogramOpts{
```
Comment (ritbl) [inline server_metrics.go:69]: why it was moved here?

File: server_metrics.go [hunk 3]
```diff
@@ -62,15 +62,22 @@ func NewServerMetrics(counterOpts ...CounterOption) *ServerMetrics {
...
+		Name:    "grpc_server_handling_seconds",
+		Help:    "Histogram of response latency (seconds) of gRPC that had been application-level handled by the server.",
+		Buckets: prom.DefBuckets,
 	}
+	for _, o := range opts {
+		o(&serverHandledHistogramOpts)
+	}
+	m.serverHandledHistogram = prom.NewHistogramVec(
```
Comment (ritbl) [inline server_metrics.go:77]: why it was moved here?

File: util.go [hunk 4]
```diff
@@ -19,15 +18,6 @@ const (
 	BidiStream   grpcType = "bidi_stream"
 )
 
-var (
```
Comment (ritbl) [inline util.go:22]: we don't need to remove this

File: server_metrics.go (hunk 3 above)
Comment (YaroslavPodorvanov) [reply server_metrics.go:77]: because in other place this useless

File: util.go (hunk 4 above)
Comment (YaroslavPodorvanov) [reply util.go:22]: we remove pre initialization so this is unused now

File: server_metrics.go (hunk 3 above)
Comment (ritbl) [reply server_metrics.go:77]: it should be in ctor as it is initialization should be there.

cons:
 - potential nil deref
 - each time it is called, it overrides serverHandledHistogram

the only pro - saving several bytes of memory.

File: util.go (hunk 4 above)
Comment (ritbl) [reply util.go:22]: I understand, but we can ignore this part

File: server_metrics.go (hunk 3 above)
Comment (YaroslavPodorvanov) [reply server_metrics.go:77]: > each time it is called, it overrides serverHandledHistogram

`serverHandledHistogram` initialize only once

**Pull request merged**